### PR TITLE
Homepage oAuth

### PR DESCRIPTION
**Pull request by yasminhillis**
Adds the oAuth to the homepage

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "build": "webpack && babel src --out-dir lib --copy-files",
     "start": "node ./src/server/",
     "dev:build": "webpack --watch",
-    "dev:start": "nodemon --exec babel-node ./src/server/",
+    "dev:start": "nodemon --exec node ./src/server/",
```
Comment (yahyaHB) [inline package.json]: can u pls explain this

File: package.json [hunk 2]
```diff
@@ -17,12 +17,9 @@
   "license": "ISC",
   "dependencies": {
     "babel-cli": "^6.26.0",
-    "babel-core": "^6.26.0",
```
Comment (yahyaHB) [inline package.json]: why did u delete it?

File: src/client/components/home/index.js [hunk 3]
```diff
@@ -20,7 +20,8 @@ class Home extends Component {
             <div className='home__text'>Founders & Coders
               sustainable opportunity-based Community</div>
           </div>
-          <input className='home__btn' type='submit' value='Sign in with' />
+          <a className='home__btn' href="auth/github">Sign in with</a>
+          // <input className='home__btn' href="/auth/github" type='submit' value='Sign in with' />
```
Comment (yahyaHB) [inline src/client/components/home/index.js]: remove the commented lines

File: src/server/index.js [hunk 4]
```diff
@@ -3,16 +3,76 @@ const path = require('path');
...
+passport.use(new Strategy({
+
+  clientID: process.env.GITHUB_CLIENTID,
+  clientSecret: process.env.GITHUB_SECRET,
+  callbackURL: 'http://localhost:3000/auth/github/callback',
+  profileFields: ['email','displayName','profileUrl','picture.type(large)']
+},(accessToken,refreshToken,profile,done)=>{
+  getUserData.github_id(profile._json.id,(err,userObj)=>{
+    if(err) {
+      return done(err)
```
Comment (yahyaHB) [inline src/server/index.js]: u can return a clear error message

File: src/server/index.js [hunk 5]
```diff
@@ -3,16 +3,76 @@ const path = require('path');
...
+    }
+
+    if(userObj.length === 0){
+  postGithubData.users(profile._json.id, profile._json.name, profile._json.email, profile._json.avatar_url, 'true' , profile._json.url,(err,userObj)=>{
+    if (err){
+      console.log(err);
```
Comment (yahyaHB) [inline src/server/index.js]: no console.log
instead u can return res.send(message of the error)

File: src/server/models/database/dbConnection.js [hunk 6]
```diff
@@ -4,5 +4,6 @@ require('env2')('./config.env');
 if (!process.env.DATABASE_URL) {
   throw new Error('DB url not found ');
 } else {
+  console.log(process.env.DATABASE_URL);
```
Comment (yahyaHB) [inline src/server/models/database/dbConnection.js]: remove console

File: package.json (hunk 2 above)
Comment (yasminhillis) [reply package.json]: I replaced its place. it's in the devDependencies now.

File: package.json (hunk 1 above)
Comment (yahyaHB) [reply package.json]: alright

**Comment by astroash** [comment]
Can you add db changes discussed this morning 😊

**Comment by yasminhillis** [comment]
@astroash  Done :+1:

File: src/server/models/database/build.sql [hunk 7]
```diff
@@ -1,22 +1,21 @@
...
     github_id BIGINT UNIQUE NOT NULL,
-    name VARCHAR(100) NOT NULL,
+    username TEXT NOT NULL,
     email VARCHAR(300) ,
     avatar VARCHAR(8000) NOT NULL,
-    newUser BOOLEAN DEFAULT false NOT NULL,
+    new_user BOOLEAN DEFAULT false NOT NULL,
     profile_url VARCHAR(8000) NOT NULL,
-	"role"  BOOLEAN DEFAULT false NOT NULL,
+	  is_admin  BOOLEAN DEFAULT false NOT NULL,
```
Comment (astroash) [inline src/server/models/database/build.sql]: This has extra indentation which makes the schema harder to read. Can this be inline with the rest of the text?

**Comment by MahmoudMH** [comment]
I am sorry, I will add comments because I have to learn what this does, but not now maybe later

**Comment by yasminhillis** [comment]
Changes made. Hence the new review request :sparkles:

**Pull request closed without merging**